packages/server/graphql/mutations/helpers/softDeleteUser.ts

```diff
@@ -0,0 +1,71 @@
+import removeSlackAuths from './removeSlackAuths'
+import removeFromOrg from './removeFromOrg'
+import segmentIo from '../../../utils/segmentIo'
+import {DataLoaderWorker} from '../../graphql'
+import removeGitHubAuth from '../../../postgres/queries/removeGitHubAuth'
+import removeAtlassianAuth from '../../../postgres/queries/removeAtlassianAuth'
+import getRethink from '../../../database/rethinkDriver'
+import TeamMemberId from 'parabol-client/shared/gqlIds/TeamMemberId'
+import parabolFetch from '../../parabolFetch'
+
+const removeGitHubAuths = async (userId: string, teamIds: string[]) =>
+  Promise.all(teamIds.map((teamId) => removeGitHubAuth(userId, teamId)))
+
+const removeAtlassianAuths = async (userId: string, teamIds: string[]) =>
+  Promise.all(teamIds.map((teamId) => removeAtlassianAuth(userId, teamId)))
+
+const softDeleteUser = async (
+  userIdToDelete: string,
+  dataLoader: DataLoaderWorker,
+  validReason?: string
+) => {
+  const r = await getRethink()
+  const orgUsers = await dataLoader.get('organizationUsersByUserId').load(userIdToDelete)
+  const orgIds = orgUsers.map((orgUser) => orgUser.orgId)
+
+  await Promise.all(
+    orgIds.map((orgId) => removeFromOrg(userIdToDelete, orgId, undefined, dataLoader))
+  )
+  const teamMemberIds = await r
+    .table('TeamMember')
+    .getAll(userIdToDelete, {index: 'userId'})
+    .getField('id')
+    .coerceTo('array')
+    .run()
+  const teamIds = teamMemberIds.map((id) => TeamMemberId.split(id).teamId)
+
+  const [parabolPayload] = await Promise.all([
+    parabolFetch(
```

Comment:
reusing the hubspot helper function isn't great. it doesn't reuse the dataloader or auth token & the error message isn't correct.

let's just call executeGraphQL like https://github.com/ParabolInc/parabol/blob/c2112010eae3cc48a47654ff467515401d722976/packages/server/email/ServerEnvironment.ts#L37-L42